src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -16,6 +16,68 @@
   <Import Project="$(BuildNumberTarget)" Condition="Exists('$(BuildNumberTarget)')" />
 
   <!-- #################################### -->
+  <!-- Version Info -->
+  <!-- #################################### -->
+
+  <ItemGroup>
+    <VersionInfoFile Include="$(BinDir)VersionInfo.json"/>
```

Comment:
If you were going to go as far as writing it to a file you should just write it to a props file that can be directly imported.